core/services/ocr2/plugins/median/services.go

```diff
@@ -67,19 +68,26 @@ func NewMedianServices(ctx context.Context,
 	}
 	spec := jb.OCR2OracleSpec
 
-	provider, err := relayer.NewMedianProvider(ctx, types.RelayArgs{
+	provider, err := relayer.NewPluginProvider(ctx, types.RelayArgs{
 		ExternalJobID: jb.ExternalJobID,
 		JobID:         spec.ID,
 		ContractID:    spec.ContractID,
 		New:           isNewlyCreatedJob,
 		RelayConfig:   spec.RelayConfig.Bytes(),
+		ProviderType:  string(spec.PluginType),
 	}, types.PluginArgs{
 		TransmitterID: spec.TransmitterID.String,
 		PluginConfig:  spec.PluginConfig.Bytes(),
 	})
 	if err != nil {
 		return
 	}
+
+	medianProvider, ok := provider.(types.MedianProvider)
+	if !ok {
+		return nil, errors.New("could not coerce PluginProvider to MedianProvider")
```

Comment:
is it better to panic? is there anyway to recover this error without restarting the node?

Reply:
It might be good to let other unaffected jobs keep running. 

And it might actually be necessary to allow the node to start up so the UI can be used to reconfigure a jobspec? :thinking: